epochStart/metachain/legacySystemSCs.go

```diff
@@ -587,6 +598,12 @@ func (s *legacySystemSCProcessor) updateMaxNodes(validatorsInfoMap state.ShardVa
 		return err
 	}
 
+	if !s.enableEpochsHandler.IsFlagEnabled(common.StakingV4StartedFlag) {
+		if maxNumberOfNodes < prevMaxNumberOfNodes {
+			return epochStart.ErrInvalidMaxNumberOfNodes
+		}
+	}
+
 	if s.enableEpochsHandler.IsFlagEnabled(common.StakingQueueFlag) {
```

Comment:
It is mandatory to be there, that flag says:
- if there is still a staking queue (so staking v4 not started)
- stake nodes from queue
We don't want to have anything to do with the queue once stakingV4 is started

Reply:
after a closer inspection, it looks like the 
s.enableEpochsHandler.IsFlagEnabled(common.StakingQueueFlag) equals !s.enableEpochsHandler.IsFlagEnabled(common.StakingV4StartedFlag)
it could be refactored so that only the StakingV4StartedFlag is used

Reply:
Removed `StakingQueueFlag`